listings/applications/erc721/src/tests.cairo

```diff
@@ -0,0 +1,90 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use starknet::testing::ContractState;
+    use starknet::ContractAddress;
+    use starknet::get_caller_address;
+
+    #[test]
+    fn test_constructor() {
+        let contract_state = ContractState::default();
+        let contract = ERC721Contract::constructor(&contract_state, "MyToken".into(), "MTK".into());
+
+        assert_eq!(contract.get_name(), "MyToken".into());
+        assert_eq!(contract.get_symbol(), "MTK".into());
+    }
+
+    #[test]
+    fn test_mint() {
+        let contract_state = ContractState::default();
+        let mut contract = ERC721Contract::constructor(&contract_state, "MyToken".into(), "MTK".into());
+
+        let to = ContractAddress::from(1);
+        let token_id = 1.into();
+
+        contract._mint(to, token_id);
+
+        assert_eq!(contract.owner_of(token_id), to);
+        assert_eq!(contract.balance_of(to), 1.into());
```

Comment:
The tests should assert that appropriate events have been emitted too.

Below are the links on how to do that depending on the testing library you u se:
- testing events in plain Cairo: https://starknet-by-example.voyager.online/getting-started/testing/contract-testing.html#testing-events
- testing events with Foundry: https://foundry-rs.github.io/starknet-foundry/testing/testing-events.html